tests/main/interfaces-mount-control/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Test for the mount-control interface
+
+environment:
+    MOUNT_SRC: /var/tmp/test-snapd-mount-control
+    SNAP_COMMON: /var/snap/test-snapd-mount-control/common
+    SNAP_NAME: test-snapd-mount-control
+
+prepare: |
+    mkdir -p "$MOUNT_SRC/dir1"
+    echo "Something" > "$MOUNT_SRC/file1"
+
+restore: |
+    rm connect_error.log
+    rm -rf "$MOUNT_SRC"
+
+execute: |
+    MOUNT_DEST="$SNAP_COMMON/target"
+
+    echo "First verify that a snap with a malicious manifest cannot be connected"
+    "$TESTSTOOLS"/snaps-state install-local test-mount-control-invalid
+    snap connect test-mount-control-invalid:mntctl 2> connect_error.log || true
+    if os.query is-trusty; then
+        echo "On Trusty, we should fail anyway due to systemd being too old"
+        MATCH "systemd version 204 is too old" < connect_error.log
+        exit 0
+    fi
+
+    MATCH 'mount-control "where" pattern is not clean' < connect_error.log
+
+    echo "Installing the test snap"
+
+    "$TESTSTOOLS"/snaps-state install-local "${SNAP_NAME}"
+
+    echo "Connecting the mount-control interface"
+    snap connect "${SNAP_NAME}":mntctl
+
+    echo "Verify that the snap can perform a mount"
+    mkdir -p "$MOUNT_DEST"
+    "${SNAP_NAME}".cmd mount -o bind,rw "$MOUNT_SRC" "$MOUNT_DEST"
+
+    echo "Verify that the mount has been performed"
+    "${SNAP_NAME}".cmd grep "$MOUNT_DEST" /proc/self/mountinfo
```

Comment:
can we also check that the mount performed in the mount namespace is not present outside the snap mount namespace?

Reply:
Added.